torch/_inductor/ir.py

```diff
@@ -684,6 +684,26 @@ def create(
         reduction_hint: ReductionHint = ReductionHint.DEFAULT,
     ):
         reduction_numel = V.graph.sizevars.simplify(sympy_product(reduction_ranges))
+
+        if reduction_numel == 0:
+            rtypes_to_inits = {
+                'sum' : 0,
```

Comment:
this needs to be `0.0` otherwise LLVM throws up: 
```
Intrinsic has incorrect return type!
i32 (i32)* @llvm.sqrt.i32
in function kernel_0d1
LLVM ERROR: Broken function found, compilation aborted!
```

possibly an issue in triton?

@ngimel